### PR TITLE
Add an integration/internal/container helper package 

### PR DESCRIPTION
**Pull request by vdemeester**
On top of #36265 

To help creating/running/… containers using the client for test integration.
This should make test more readable and reduce duplication a bit.

Usage example

```go
// Create a default container named foo
id1 := container.Create(t, ctx, client, container.WithName("foo"))
// Run a default container with a custom command
id2 := container.Run(t, ctx, client, container.WithCmd("echo", "hello world"))
```

This replace `runSimpleContainer` and `createSimpleContainer` 👼 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: integration/internal/container/container.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+type TestContainerConfig struct {
+	Name             string
+	Config           *container.Config
+	HostConfig       *container.HostConfig
+	NetworkingConfig *network.NetworkingConfig
+}
+
+func Create(t *testing.T, ctx context.Context, client client.APIClient, ops ...func(*TestContainerConfig)) string {
+	config := &TestContainerConfig{
```
Comment (dnephin) [inline integration/internal/container/container.go]: Should we `t.Helper()` in these so the error gets reported from the caller?

File: integration/internal/container/container.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/require"
+)
+
+type TestContainerConfig struct {
+	Name             string
```
Comment (dnephin) [inline integration/internal/container/container.go]: I wish we had this struct (and ops) for the actual `client` package! I guess we can look at doing that later

**Review by dnephin — COMMENTED** [review]
LGTM

File: integration/internal/container/container.go (hunk 2 above)
Comment (vdemeester) [reply integration/internal/container/container.go]: That's definitely sthg we should do 👼 (It's now on my todo 😛)

**Review by yongtang — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**